spock-core/src/main/java/org/spockframework/runtime/extension/builtin/ConditionalExtension.java

```diff
@@ -108,7 +118,7 @@ public IterationCondition(Closure condition, T annotation) {
 
     @Override
     public void intercept(IMethodInvocation invocation) throws Throwable {
-      Object result = evaluateCondition(condition, invocation.getIteration().getDataVariables());
+      Object result = evaluateCondition(condition, invocation, invocation.getIteration().getDataVariables());
```

Comment:
Why do you pass the `invocation` and not the `invocation.getInstance()` here?

Reply:
Because it is done the same for `RetryConditionContext`, so I thought I just follow the same pattern here.
But I can of course also change that if you prefer.
If you do, should I change it for both?